internal/scaler/awsec2asg/aws_ec2_asg.go

```diff
@@ -326,7 +332,7 @@ func (s *Scaler) Register(ctx context.Context) error {
 
```

Comment:
Since we need to deal with the min-max logic only during a scale-out. How about we do something like this

```go
if *asg.DesiredCapacity < inputData.DesiredCount {
    // scale-out
    if s.DisableScaleOut != nil && *s.DisableScaleOut {
	    return
    }

    desiredCapacity := inputData.DesiredCount
    if desiredCapacity > *asg.MaxSize {
        desiredCapacity = *asg.MaxSize
    }

    updateAsg = &autoscaling.UpdateAutoScalingGroupInput{
	    AutoScalingGroupName: &inputData.ASGName,
	    DesiredCapacity:      &desiredCapacity,
    }
}
```

Reply:
I am thinking if we should do something similar for scale-in and `desiredCount < *asg.MinSize` :thinking:

Reply:
oops, I see. That is what your original logic does. Sorry :sweat_smile: 

```
			desiredCapacity := int64(math.Min(
				math.Max(float64(*asg.MinSize), float64(inputData.DesiredCount)),
				float64(*asg.MaxSize)))
```

Reply:
Yes..pls let me know if this works or I can update it the other way too